### ISSUE TITLE
The Normandy recipe cannot be created if the string preference value is empty

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla/normandy#2236]
[Affected versions]:
- Chrome 83.0.4103.97
- Firefox release 77;
- Firefox Beta 78.0b7;
- Nightly 79.0a1;

[Affected Platforms]:
- All Windows;
- All macOS;

[Prerequisites]:
- Have access to https://delivery-console.stage.mozaws.net

[Steps to reproduce]:
1. Open the browser and navigate to "https://delivery-console.stage.mozaws.net"
2. Create a multi preference experiment.
3. Fill all the mandatory fields with valid data.
4. Create a branch with a string preference and leave the "Preference Value" field empty.
5. Click the "Save" button.
6. Observe the behavior.

[Expected result]:
- The recipe is successfully saved.

[Actual result]:
- The recipe cannot be saved and the "preferenceValue This field may not be blank." error is displayed.

[Notes]:
- Here is the request as a HAR file: [link](https://drive.google.com/open?id=1AiBuCOgmG68cgOjXDPrZuGgNS9LlLpVs).
- Attached a screen recording with the issue:
![string preference value empty](https://user-images.githubusercontent.com/19226534/84676353-15611480-af36-11ea-92a1-5890cf1d5b24.gif)

**Comment by mythmon** [comment]
As a side note, @SoftVision-CosminMuntean I would suggest using [Normandy  Devtools](https://mozilla-extensions.github.io/normandy-devtools/#/) for future testing. It supports all environments, and it is has a better chance of being  updated than Delivery Console going forward. It is still lacking history support,  but most everything else still works.

**Comment by shell1** [comment]
@mythmon how does this issue impact entering "" in control branch.  Does it mean it doesn't work?  Or just that we can't simulate the change with existing tools?

**Comment by mythmon** [comment]
It means it does not work, we cannot instruct Normandy to set that as a branch value. I've recommended using a dummy branch for this study. Note that Experiment also does not support this.

**Comment by shell1** [comment]
Is a dummy branch what you said where Normandy can group the people and NOT DO ANYTHING to them.  which isn't a true control - in that we aren't enforcing that they are in the state we want.

But in this case it sounds like the best option for gathering a set amount of random users NOT in the treatment to compare against.

**Comment by mythmon** [comment]
> Is a dummy branch what you said where Normandy can group the people and NOT DO ANYTHING to them

Not quite. We still have to set some preference, but we can make it an ineffectual one like `test.the-experiment.blah=1`. We have to change something.

**Comment by shell1** [comment]
doh, of course.  thanks @mythmon   That is an acceptable work-around - that gets us most of the benefit of a random control (with a caveat that it's not guaranteed "controlled"), but won't trigger needing more testing as we've tested doing this.  

do you want me to adapt the experimenter ticket for the control to be tile-control-branch | 1?  

that seems like a safe, but clear one to add.  I'm going off the pattern used in https://experimenter.services.mozilla.com/experiments/aaaaa-test/

**Comment by SoftVision-CosminMuntean** [comment]
I have verified this issue and the Normandy recipe can be created if the string preference value is empty in the delivery console. In order to verify this I have used the following scenarios:
- Verified that the string preferences can be saved if the value is empty.
- Verified that the integer preferences can be saved if the value is empty.
- Verified that you are successfully enrolled in an experiment that has an integer and string preferences with an empty value.
- Verified that the preferences are correctly displayed in about:config page after enrollment.

I have verified the delivery console part on Chrome 84.0.4147 and Firefox 80.0 release on Windows 10 and verified the enrollment part on Firefox 80, Beta 81.0b4, and the latest Nightly 82.0a1 on Windows 10 x64, macOS 10.15.6 and Ubuntu 20.04 x64.
